core/common/io/src/main/java/org/eclipse/rdf4j/common/lang/FileFormat.java

```diff
@@ -97,9 +97,12 @@ public FileFormat(String name, String mimeType, Charset charset, Collection<Stri
 	public FileFormat(String name, Collection<String> mimeTypes, Charset charset, Collection<String> fileExtensions) {
 		assert name != null : "name must not be null";
 		assert mimeTypes != null : "mimeTypes must not be null";
-		assert !mimeTypes.isEmpty() : "mimeTypes must not be empty";
 		assert fileExtensions != null : "fileExtensions must not be null";
 
+		if (mimeTypes.isEmpty()) {
```

Comment:
I know the bug report was only about the mime types, but perhaps we should do this for the other arguments as well - what do you think? For the single object inputs we could just use a `Objects.requireNonNull` and for the other collection parameter (file extensions) we could do a similar check to what's you've added here.

Reply:
Quoting from #1090 HansBrende's original issue 

> Note: the 2 other assert statements requiring the collections to be non-null do not pose the same problem, because the null-check is handled down the line by List.addAll(collectionThatIsNotNull).

Therefore I did not go ahead with adding checks for the same. 

However, I do think we can add `Objects.requireNonNull` for the single inputs instead of differing it to  `List.addAll()` and a similar check for the file extensions parameter. Let me know if you would like me to go ahead and implement these.

Reply:
Ah good catch. Actually let's keep things simple for now, and leave as-is. Thanks :)